src/device-state/current-state.ts

```diff
@@ -2,55 +2,47 @@ import * as _ from 'lodash';
 import * as url from 'url';
 import { CoreOptions } from 'request';
 
-import * as constants from '../lib/constants';
+import { backoffPromise } from '../lib/backoff';
 import { log } from '../lib/supervisor-console';
 import { InternalInconsistencyError, StatusError } from '../lib/errors';
 import { getRequestInstance } from '../lib/request';
 import * as sysInfo from '../lib/system-info';
+import sleep from '../lib/sleep';
 
 import { DeviceStatus } from '../types/state';
 import * as config from '../config';
-import { SchemaTypeKey, SchemaReturn } from '../config/schema-type';
 import * as eventTracker from '../event-tracker';
 import * as deviceState from '../device-state';
 
-// The exponential backoff starts at 15s
-const MINIMUM_BACKOFF_DELAY = 15000;
-
 const INTERNAL_STATE_KEYS = [
 	'update_pending',
 	'update_downloaded',
 	'update_failed',
 ];
 
 export let stateReportErrors = 0;
+
 const lastReportedState: DeviceStatus = {
 	local: {},
 	dependent: {},
 };
-const stateForReport: DeviceStatus = {
```

Comment:
Keeping a local copy of state for report didn't make sense because it would become outdated theoretically right away if something changes. The other types didn't seem necessary either.

Reply:
It doesn't seem necessary but it does predate this separated reporting module even so I'd be a little wary of changing it without ensuring it's safe and testing it. I think it makes sense to remove if it's unnecessary but I'd lean towards delaying it for the moment since if it does break something it's likely to do it in a way that puts unnecessary load on the API, and removing it is not urgent/necessary other than to clean up the code